packages/ai-jsx/src/batteries/constrained-output.tsx

```diff
@@ -78,9 +79,19 @@ export type TypedObjectCompletionWithRetry = TypedObjectCompletion & { retries?:
  *    Intermediate results that are valid, are also yielded.
  */
 export async function* JsonChatCompletion(
-  props: Omit<TypedObjectCompletionWithRetry, 'typeName' | 'parser' | 'partialResultCleaner'>,
+  { schema, ...props }: Omit<TypedObjectCompletionWithRetry, 'typeName' | 'parser' | 'partialResultCleaner'>,
   { render }: AI.ComponentContext
 ) {
+  if (schema) {
+    try {
+      // TODO: do not merge with this. This screws up progressive loading.
+      return yield* render(<JsonChatCompletionFunctionCall schema={schema} {...props} />);
```

Comment:
As the note says, we shouldn't merge until we figure this part out.

Reply:
Does this screw up progressive loading because chat completion doesn't stream partials of a `<FunctionCall />` or is there something else going on here?